### PR TITLE
Implement and test IntoIterator

### PR DESCRIPTION
**Pull request by FintanH**

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -621,6 +621,15 @@ impl<T> From<(T, Vec<T>)> for NonEmpty<T> {
     }
 }
 
+impl<T> IntoIterator for NonEmpty<T> {
+    type Item = T;
+    type IntoIter = std::iter::Chain<std::iter::Once<T>, std::vec::IntoIter<Self::Item>>;
```
Comment (cloudhead) [inline src/lib.rs]: Can we import some of this stuff to make this line more readable? Maybe we import at least `std::{iter, vec}` at the top level?

**Comment by cloudhead** [comment]
Perhaps `impl<T> Into<Vec<T>> for NonEmpty<T>` can use this too.

**Comment by FintanH** [comment]
@cloudhead: Speaking of `Into`, apparently [you shouldn't implement it](https://rust-lang.github.io/api-guidelines/interoperability.html#conversions-use-the-standard-traits-from-asref-asmut-c-conv-traits)

**Comment by cloudhead** [comment]
@FintanH yes true, we should turn it into a `impl<T> From<NonEmpty<T>> for Vec<T>`

**Pull request merged**